8288443: Simplify vmClasses::resolve_all()

iklam opened the pull request:
Please review this simple clean up:

- There's no need to initialize the JSR 292 classes separately (the `-XX:+EnableInvokeDynamic` option has been removed long time ago).
- Separate initialization of the reference classes is not necessary when CDS is enabled.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8288443](https://bugs.openjdk.org/browse/JDK-8288443): Simplify vmClasses::resolve_all()


### Reviewers
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [e802861c](https://git.openjdk.org/jdk/pull/9157/files/e802861cfa26763f0be9f625aa706a06c7a29847)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [e802861c](https://git.openjdk.org/jdk/pull/9157/files/e802861cfa26763f0be9f625aa706a06c7a29847)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9157/head:pull/9157` \
`$ git checkout pull/9157`

Update a local copy of the PR: \
`$ git checkout pull/9157` \
`$ git pull https://git.openjdk.org/jdk pull/9157/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9157`

View PR using the GUI difftool: \
`$ git pr show -t 9157`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9157.diff">https://git.openjdk.org/jdk/pull/9157.diff</a>

</details>

calvinccheung reviewed (approved):
Looks good.

dholmes-ora reviewed (approved):
Seems fine.

There are related comments about ordering and 1.4 JDKs in vmClassMacros.hpp that should also be cleaned up. I find it interesting it says

 /* Note: MethodHandle must be first, and VolatileCallSite last in group */ 

when DirectMethodHandle is actually first.

Thanks.

iklam commented:
> Seems fine.
> 
> There are related comments about ordering and 1.4 JDKs in vmClassMacros.hpp that should also be cleaned up. I find it interesting it says
> 
> /* Note: MethodHandle must be first, and VolatileCallSite last in group */
> 
> when DirectMethodHandle is actually first.
> 
> Thanks.

I removed the out-dated comments about supporting earlier JDKs. Also removed the info about the ordering of the `MethodHandle` classes. I have no idea whether or why the ordering is important, but this mystery should be covered by the comment about the `VM_CLASSES_DO` macro definition.

```
// The order of these definitions is significant: the classes are
// resolved by vmClasses::resolve_all() in this order. Changing the
// order may require careful restructuring of the VM start-up sequence.
```

dholmes-ora reviewed (approved):
Thanks for the update

iklam commented:
Thanks @calvinccheung @dholmes-ora @coleenp for the review.
Passed tiers 1-4.

/integrate
